### PR TITLE
/se/ameba: Add RTL SE Secure Storage Write Protect

### PR DESCRIPTION
**Pull request by ziyik**
- Add SE Secure Storage Write Protect for Flash first 512K bytes
- Flash Write protect will be enable when DA AES key is written in efuse

### CODE REVIEW

File: os/se/ameba/rtl_security_api_wrapper_tz.c
Comment (jeongchanKim) [inline os/se/ameba/rtl_security_api_wrapper_tz.c:74]: Is there any use case that calling `FLASH_SetStatusBits()` without `FLASH_Write_Lock()`?
If there is no, how about adding lock and unlock inside the `FLASH_SetStatusBits()`? Then we can reduce the code.
When I checked, all `FLASH_SetStatusBits` are binded with `FLASH_Write_Lock` and `FLASH_Write_Unlock`.
And many functions related to `FLASH_XXX` have lock and unlock inside.

Comment (ziyik) [reply os/se/ameba/rtl_security_api_wrapper_tz.c:74]: FLASH_SetStatusBits() is a API located in ROM, we cannot modify it.

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Comment by ziyik** [comment]
@jeongchanKim How do set do not merge label? This PR cannot be merge before [#4787](https://github.com/Samsung/TizenRT/pull/4787)

**Comment by jeongchanKim** [comment]
> @jeongchanKim How do set do not merge label? This PR cannot be merge before [#4787](https://github.com/Samsung/TizenRT/pull/4787)

As I know, only maintainer can attach the label maybe...
So if you want to mark, Please add [DO NOT MERGE] at the PR title.

**Comment by ziyik** [comment]
> > @jeongchanKim How do set do not merge label? This PR cannot be merge before [#4787](https://github.com/Samsung/TizenRT/pull/4787)
> 
> As I know, only maintainer can attach the label maybe...
> So if you want to mark, Please add [DO NOT MERGE] at the PR title.

Okay, thank you

**Comment by sunghan-chang** [comment]
@ziyik 
1. Please resolve the conflicts.
```
Conflicting files
build/tools/amebad/gnu_utility/km4_image3_all.bin
build/tools/amebad/gnu_utility/km4_image3_all_fpu.bin
build/tools/amebad/gnu_utility/km4_image3_psram.bin
build/tools/amebad/gnu_utility/km4_image3_psram_fpu.bin
os/se/ameba/rtl_security_api_wrapper_tz.c
```
2. PR #4787 was merged. Can I merge this?
3. Could you write the condition about write protection working?

**Comment by ziyik** [comment]
> @ziyik
> 
> 1. Please resolve the conflicts.
> 
> ```
> Conflicting files
> build/tools/amebad/gnu_utility/km4_image3_all.bin
> build/tools/amebad/gnu_utility/km4_image3_all_fpu.bin
> build/tools/amebad/gnu_utility/km4_image3_psram.bin
> build/tools/amebad/gnu_utility/km4_image3_psram_fpu.bin
> os/se/ameba/rtl_security_api_wrapper_tz.c
> ```
> 
> 1. PR #4787 was merged. Can I merge this?
> 2. Could you write the condition about write protection working?

I'm rebase-ing this PR.
Once it is done it can be merge, will write the condition about write protection working as well.

Thanks

**Comment by ziyik** [comment]
@sunghan-chang I have rebase this PR, you can merge it.
I will write the condition about write protection working at the description of PR.

Thanks

**Comment by sunghan-chang** [comment]
@ziyik 1. Could you write description in commit? not in PR?
```sh
1. git commit --amend
2. Add below in commit
    Add SE Secure Storage Write Protect for Flash first 512K bytes
    Flash Write protect will be enable when DA AES key is written in efuse
3. git push -f
```
2. This PR does not include updating .a (or .bin) anymore. Previously it did.
    Is this right? Does it mean the same changes on .a or .bin were included in another PR and it was merged already?

**Comment by sunghan-chang** [comment]
```
build/tools/amebad/gnu_utility/km4_image3_all.bin
build/tools/amebad/gnu_utility/km4_image3_all_fpu.bin
build/tools/amebad/gnu_utility/km4_image3_psram.bin
build/tools/amebad/gnu_utility/km4_image3_psram_fpu.bin
```
These files

**Comment by ziyik** [comment]
> @ziyik 1. Could you write description in commit? not in PR?
> 
> ```shell
> 1. git commit --amend
> 2. Add below in commit
>     Add SE Secure Storage Write Protect for Flash first 512K bytes
>     Flash Write protect will be enable when DA AES key is written in efuse
> 3. git push -f
> ```
> 
> 1. This PR does not include updating .a (or .bin) anymore. Previously it did.
>    Is this right? Does it mean the same changes on .a or .bin were included in another PR and it was merged already?

Yes, you are correct. but previous commit will not enable the write protect.
It is [#PR 4787](https://github.com/Samsung/TizenRT/pull/4787)

**Comment by sunghan-chang** [comment]
@ziyik Ok. thank you for fast reply. Once you update the commit description, I will merge this.

**Comment by ziyik** [comment]
> @ziyik Ok. thank you for fast reply. Once you update the commit description, I will merge this.

Done editing the comment.
Let me know if there is anything I need to modify.

Thank you

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**
